tests/test_client_session.py

```diff
@@ -467,6 +467,60 @@ async def create_connection(req, traces, timeout):
         c.__del__()
 
 
+@pytest.mark.parametrize("protocol", ["http", "https", "ws", "wss"])
+async def test_ws_connect_allowed_protocols(
+    create_session: Any,
+    create_mocked_conn: Any,
+    protocol: str,
+    ws_key: Any,
+    key_data: Any,
+) -> None:
+    url = URL(f"{protocol}://example.com")
+    req = mock.Mock()
+    req_factory = mock.Mock(return_value=req)
+    resp = mock.Mock()
+    resp.status = 101
+    resp.headers = {
+        hdrs.UPGRADE: "websocket",
+        hdrs.CONNECTION: "upgrade",
+        hdrs.SEC_WEBSOCKET_ACCEPT: ws_key,
+    }
+    resp.url = url
+    resp.cookies = SimpleCookie()
+    resp.start = mock.AsyncMock()
+    req.send = mock.AsyncMock(return_value=resp)
+    session = await create_session(request_class=req_factory)
+
+    connections = []
+    original_connect = session._connector.connect
+
+    async def connect(req, traces, timeout):
+        conn = await original_connect(req, traces, timeout)
+        connections.append(conn)
+        return conn
+
+    async def create_connection(req, traces, timeout):
+        # return self.transport, self.protocol
+        conn = create_mocked_conn()
+        return conn
+
+    session._connector.connect = connect
+    session._connector._create_connection = create_connection
+    session._connector._release = mock.Mock()
```

Comment:
This one should be using .patch.object(), right?

Reply:
68ea65fed759e6f8ae69e7146127a30e26e9552f